cloud/openstack/os_stack.py

```diff
@@ -0,0 +1,213 @@
+      default: 3600
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+EXAMPLES = '''
+---
+- name: create stack
+  ignore_errors: True
+  register: stack_create
+  os_heat:
+    name: "{{ stack_name }}"
+    state: present
+    template: "/path/to/my_stack.yaml"
+    environment:
+    - /path/to/resource-registry.yaml
+    - /path/to/environment.yaml
+    parameters:
+        os_user: {{ os_user }}
+        os_password: {{ os_password }}
+        os_tenant: {{ os_tenant }}
+        os_auth_url: {{ os_auth_url }}
+        bmc_flavor: m1.medium
+        bmc_image: CentOS
+        key_name: default
+        private_net: {{ private_net_param }}
+        node_count: 2
+        name: undercloud
+        image: CentOS
+        my_flavor: m1.large
+        external_net: {{ external_net_param }}
+'''
+
+RETURN = '''
+#
+'''
+
+def _create_stack(module, stack, cloud):
+    try:
+        stack = cloud.create_stack(module.params['name'],
+                                       template_file=module.params['template'],
+                                       environment_files=module.params['environment'],
+                                       timeout=module.params['timeout'],
+                                       wait=True,
+                                       rollback=module.params['rollback'],
+                                       **module.params['parameters'])
+
+        stack = cloud.get_stack(stack.id, None)
+        if stack.stack_status == 'CREATE_COMPLETE':
+            return stack
+        else:
+            return False
+            module.fail_json(msg = "Failure in creating stack: ".format(stack))
+    except shade.OpenStackCloudException as e:
+        module.fail_json(msg=e.message)
+
+def _update_stack(module, stack, cloud):
+    try:
+        stack = cloud.update_stack(
+            module.params['name'],
+            template_file=module.params['template'],
+            environment_files=module.params['environment'],
+            timeout=module.params['timeout'],
+            rollback=module.params['rollback'],
+            wait=True)
+
+        if stack['stack_status'] == 'UPDATE_COMPLETE':
+            return stack
+        else:
+            module.fail_json(msg = "Failure in updating stack: %s" %
+                             stack['stack_status_reason'])
+    except shade.OpenStackCloudException as e:
+        module.fail_json(msg=e.message)
+
+def _system_state_change(module, stack, cloud):
+    state = module.params['state']
+    if state == 'present':
+        if not stack:
+            return True
+    if state == 'absent' and stack:
+        return True
+    return False
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        template=dict(default=None),
+        environment=dict(default=None, type='list'),
+        parameters=dict(default={}, type='dict'),
+        rollback=dict(default=False),
+        timeout=dict(default=3600, type='int'),
```

Comment:
'timeout' is already defined in:

    https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/openstack.py#L85